src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/ReadAndUpdateOperation.java

```diff
@@ -64,10 +68,39 @@ public Uni<Supplier<CommandResult>> execute(
             .getDocuments(dataApiRequestInfo, queryExecutor, findOperation().pageState(), null);
     return docsToUpdate
         .onItem()
-        .transformToMulti(
+        .transformToUni(
             findResponse -> {
               pageStateReference.set(findResponse.pageState());
               final List<ReadDocument> docs = findResponse.docs();
+
+              // vectorize the updateClause or ReplacementDocument as needed
+              Uni<Boolean> vectorization = Uni.createFrom().item(false);
+              final DataVectorizer dataVectorizer =
+                  dataVectorizerService.constructDataVectorizer(dataApiRequestInfo, commandContext);
+
+              // 1. UpdateCommand(updateOne, findOneAndUpdate, updateMany):
+              if (documentUpdater.updateType() == DocumentUpdater.UpdateType.UPDATE) {
```

Comment:
move this whole test into the documentUpdater, make it handle the empty list, and use isEmpty when testing if a list is empty

e.g. DocumentUpdater.hasVectorizeDiff(docs, bool isUpsert). 

if docs list empty if the matchedCount  is zero do not need to pass, otherwise will need to pass the match count as well